docs/principles/testing.md

```diff
@@ -0,0 +1,45 @@
+# Testing Strategy
+
+## End-to-End Testing
+
+End-to-end testing tests an entire product, service or application. Individual products should each have their own tests.
```

Comment:
Should this mention that external dependencies (such as Azure search) aren't under test so should be mocked out? I'd anticipate that this will keep coming up as a question about our automated end-to-end tests.

Suggest something like the following:

> Where possible, external dependencies should be mocked out. This is because external dependencies can slow down these tests and make them brittle. Testing with external dependencies should be carried out manually.